projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -5619,10 +5647,14 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         if (this.dataView.slice(rowIndex, rowIndex + 1).find(rec => rec.expression || rec.childGridsData)) {
             visibleColIndex = -1;
         }
-        const shouldScrollVertically = this.navigation.shouldPerformVerticalScroll(rowIndex, visibleColIndex);
+        // If the target row is pinned no need to scroll as well.
+        const shouldScrollVertically =
+            !this.isRecordPinnedByIndex(rowIndex) && this.navigation.shouldPerformVerticalScroll(rowIndex, visibleColIndex);
         const shouldScrollHorizontally = this.navigation.shouldPerformHorizontalScroll(visibleColIndex, rowIndex);
         if (shouldScrollVertically) {
-            this.navigation.performVerticalScrollToCell(rowIndex, visibleColIndex,
+            // Only for top pinning we need to subtract pinned count because virtualization indexing doesn't count pinned rows.
+            const scrollRowIndex = this.isRowPinningToTop ? rowIndex - this.pinnedRecordsCount : rowIndex;
+            this.navigation.performVerticalScrollToCell(scrollRowIndex, visibleColIndex,
```

Comment:
As performVerticalScrollToCell is called in other places (hierarchical grid and mrl nav services) the scroll row index should be resolved inside the method itself to ensure consitency.

Reply:
Also most calculations seem to be using the pinnedRecordsCount, which may return incorrect results if some of the pinned records are filtered (as pinnedRecordsCount is not affected by data operations).